packages/notifications/push/amplify_push_notifications/android/src/main/kotlin/com/amazonaws/amplify/amplify_push_notifications/PushNotificationUtils.kt

```diff
@@ -63,24 +60,35 @@ fun NotificationPayload.asChannelMap(): Map<String, Any?> {
         "goToUrl" to this.action[PushNotificationsConstants.PINPOINT_URL],
         "deeplinkUrl" to this.action[PushNotificationsConstants.PINPOINT_DEEPLINK],
         "fcmOptions" to mapOf(
-            "senderId" to senderId,
-            "messageId" to messageId,
-            "sentTime" to sendTime
-            // TODO: add channelId if needed
+            "senderId" to senderId, "messageId" to messageId, "sentTime" to sendTime
+            // TODO(Samaritan1011001): add channelId if needed
         ),
         "data" to this.rawData
     )
 }
 
-fun Bundle.asPayloadFromExtras(): NotificationPayload? {
-    return this.getBundle(PAYLOAD_KEY)?.toNotificationsPayload()
+val Intent.isPushNotificationIntent: Boolean
+    get() = extras?.containsKey("google.message_id") ?: false
```

Comment:
I think this is misleading since it wouldn't handle the intents from our created notifications correctly. So universally extending Intent like this is probably going to cause issues for future contributors to this project

Reply:
Yeah, that check is not actually needed. So removing it since it is only called from `onMessageReceived` .